src/panel/category/CategoryPanel.jsx

```diff
@@ -32,24 +32,31 @@ export default class CategoryPanel extends React.Component {
   }
 
   componentDidMount() {
-    const { category } = this.props.poiFilters;
-
     this.mapMoveHandler = listen('map_moveend', this.fetchData);
-
-    if (category) {
-      Telemetry.add(Telemetry.POI_CATEGORY_OPEN, null, null, { category });
-      const { label } = CategoryService.getCategoryByName(category);
-      SearchInput.setInputValue(label.charAt(0).toUpperCase() + label.slice(1));
-    }
-
     window.execOnMapLoaded(() => { this.fitMapAndFetch(); });
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps) {
+    const { bbox } = this.props;
+    const { category } = this.props.poiFilters;
+
     const panelContent = document.querySelector('.panel-content');
     if (panelContent) {
       panelContent.scrollTop = 0;
     }
+
+    if (category !== prevProps.poiFilters.category) {
+      if (category) {
+        Telemetry.add(Telemetry.POI_CATEGORY_OPEN, null, null, { category });
+        const { label } = CategoryService.getCategoryByName(category);
+        SearchInput.setInputValue(label.charAt(0).toUpperCase() + label.slice(1));
+      }
+      this.fetchData();
+    }
+
+    if (bbox !== prevProps.bbox) {
+      this.fitMapAndFetch();
```

Comment:
You probably need to wrap it in a `window.execOnMapLoaded` call like it's done in the didMount to avoid errors or calls without effects if the user interacts with the panel while the map is still loading.
I know this is a mess, I have plans to remove that, but not yet.

Reply:
ok, done